routers/web/repo/view.go

```
@@ -728,16 +728,61 @@ func checkHomeCodeViewable(ctx *context.Context) {
 	ctx.NotFound("Home", fmt.Errorf(ctx.Tr("units.error.no_unit_allowed_repo")))
 }
 
+func checkCitationFile(ctx *context.Context) {
+	// Check if repo is not empty and if we are on repo src
+	if ctx.Repo.Repository.IsEmpty {
+		return
+	}
+	entry, _ := ctx.Repo.Commit.GetTreeEntryByPath(ctx.Repo.TreePath)
```

Don't ignore errors. Tests are failing due to this.

---

It's possible, but it's your suggestion check this:
> Why is this check here anyway? Looking at [`renderCode`](https://github.com/go-gitea/gitea/blob/9eb6c4a0dfaed86713952da19a8aa5079e97addb/routers/web/repo/view.go#L901) it assume that you can safely execute `tree, err := ctx.Repo.Commit.SubTree(ctx.Repo.TreePath)` without any necessary checks.

I'm going to roll back and check if errors will be fixed.

---

> , but it's your suggestion check this:

I was talking about removing the check altogether, not just the error check. Thereby referencing to the code at [`renderCode`](https://github.com/go-gitea/gitea/blob/9eb6c4a0dfaed86713952da19a8aa5079e97addb/routers/web/repo/view.go#L901) which is how this code should look like.

---

> > , but it's your suggestion check this:
> 
> I was talking about removing the check altogether, not just the error check. Thereby referencing to the code at [`renderCode`](https://github.com/go-gitea/gitea/blob/9eb6c4a0dfaed86713952da19a8aa5079e97addb/routers/web/repo/view.go#L901) which is how this code should look like.

Sorry but I don't understand, what checks are you talking about ?

---

> > , but it's your suggestion check this:
> 
> I was talking about removing the check altogether, not just the error check. Thereby referencing to the code at [`renderCode`](https://github.com/go-gitea/gitea/blob/9eb6c4a0dfaed86713952da19a8aa5079e97addb/routers/web/repo/view.go#L901) which is how this code should look like.

I made this check: `if ctx.Repo.Repository.IsEmpty` because if the repo is empty we are sure that we haven't `CITATION.cff` file and is useless to execute function. If you don't talking about this check, tell me what you think.